src/app/(BottomTabNavigation)/Profile/EditName/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { router } from 'expo-router';
+import React, { useEffect, useState } from 'react';
+import { Text, View, TouchableOpacity, TextInput } from 'react-native';
+
+import styles from './styles';
+import {
+  getCurrentUserInfo,
+  updateCurrUserName,
+} from '../../../../supabase/queries/auth';
+import { User, userInstance } from '../../../../types/types';
+
+function EditNameScreen() {
+  const [currSession, setCurrSession] = useState<User>(userInstance);
+  const [fullName, setFullName] = useState<string>();
+  useEffect(() => {
+    getCurrentUserInfo().then(result => {
+      setCurrSession(result);
+      setFullName(result.fullName);
+    });
+  }, []);
+  return (
+    <View>
+      <Text>Edit Name {currSession.fullName}</Text>
+      <TextInput
+        style={styles.input}
+        value={fullName}
+        onChangeText={setFullName}
+        placeholder="Full Name"
+      />
+      <TouchableOpacity
+        style={styles.button}
+        onPress={() => {
+          if (fullName) {
```

Comment:
This if statement is reached every time because `fullName` always has the user's name since we fetch it and set it at the start. I don't know how big of an issue it would be to run this query each time or if this would change with `AuthContext` but if you think running the query each time is okay, let's remove the if-else and just run both every time since that is what it is currently doing. Context: tested by logging inside if statement and reached even when name was not changed.

Reply:
I think this is something that's probably best to fix after auth context is implemented as it will be easier/less power to check if the name changed at all. But I agree that it's not the best to query each time and I need to fix.

Reply:
Okay let's just address this when we implement Auth Context then